pseudospectral/spectra/free_fermions_2D.py

```diff
@@ -1,46 +1,57 @@
 import numpy as np
+import scipy
 
+def cartisean_product(array_1, array_2):
+    return np.array([[x,y] for x in array_1 for y in array_2])
 class FreeFermions2D:
     """
     Spectrum class to represent the spectrum (eigenfunctions, eigenvalues) of the 2D free fermions operator.
     Operator = (\sigma_z \partial_{t} + \sigma_x \partial_{x}) + (m * \Identity) + (\mu * \sigma_z)
                 , where the sigmas are the Pauli matrices.
     
-    The operator is discretized on a 2D lattice with n_t lattice points in the time axis and n_x lattice points in the x axis.
+    The operator is discretized on p_0 2D lattice with n_t lattice points in the time axis and n_x lattice points in the x axis.
     We assume periodic boundary conditions in both directions with lengths L_t and L_x respectively.            
 
     Args:
         mu: Chemical potential (Fermi energy)
         m: mass parameter
         L_t: length of the system in the time axis
-        L_x: length of the system in the x axis
-        n_t: number of lattice points in the time axis
-        n_x: number of lattice points in the x axis
+        L_x: length of the system in the x axis 
+        n_t: number of lattice points in the time axis (even)
+        n_x: number of lattice points in the x axis (odd)
+
+    # array_t = [-(N-1)/2, ..., -1/2 , 1/2 , ..., (N-1)/2]
+    # array_x = [-(N-1)/2, ..., -1, 0, 1, ..., (N-1)/2]
     """
     
     def __init__(self, mu, m, L_t, L_x, n_t, n_x):
```

Comment:
```suggestion
    def __init__(self, n_t, n_x, *, mu=0, m=0, L_t=1, L_x=1):
```
Reasoning:
- `n_t` and `n_x` are always required and their order is natural.
- All other parameters should have reasonable defaults when they are not given because the user might not even be aware they exist.
- All other parameters should be named as to not confuse them (a line `1, 2, 3, 4, 5, 6` does not really tell you much about what's going on here).

Reply:
See #38